src/lib/components/workflow/workflows-summary-configurable-table/workflows-summary-configurable-table.svelte

```diff
@@ -0,0 +1,468 @@
+        >
+          {#if $supportsBulkActions}
+            <th>
+              <Checkbox
+                id="select-visible-workflows"
+                onDark
+                hoverable
+                {checked}
+                {indeterminate}
+                on:change={handleCheckboxChange}
+              />
+            </th>
+          {/if}
+          {#if $supportsBulkActions && showBulkActions}
+            <th
+              class="text-left text-sm font-medium overflow-visible whitespace-nowrap font-secondary px-2"
+              colspan={pinnedColumns.length || 1}
+            >
+              {#if allSelected}
+                <span class="font-semibold">
+                  All {filteredWorkflowCount} selected
+                </span>
+              {:else}
+                <span class="font-semibold"
+                  >{selectedWorkflowsCount} selected</span
+                >
+                <span>
+                  (or <button
+                    data-testid="select-all-workflows"
+                    on:click={handleSelectAll}
+                    class="cursor-pointer underline"
+                    >select all {filteredWorkflowCount}</button
+                  >)
+                </span>
+              {/if}
+              <div class="ml-4 inline-flex gap-2">
+                {#if cancelEnabled}
+                  <BulkActionButton
+                    testId="bulk-cancel-button"
+                    disabled={namespaceWriteDisabled}
+                    on:click={() => dispatch('cancelWorkflows')}
+                    >Request Cancellation</BulkActionButton
+                  >
+                {/if}
+                {#if terminateEnabled}
+                  <BulkActionButton
+                    variant="destructive"
+                    testId="bulk-terminate-button"
+                    disabled={namespaceWriteDisabled}
+                    on:click={() => dispatch('terminateWorkflows')}
+                    >Terminate</BulkActionButton
+                  >
+                {/if}
+              </div>
+            </th>
+          {:else}
+            {#each pinnedColumns as column}
+              <WorkflowsSummaryTableHeaderCell {column} {sortDisabled} />
+            {/each}
+          {/if}
+        </tr>
+        {#if $updating}
+          <ProgressBar />
+        {/if}
+      </thead>
+      <tbody>
+        {#if $workflowTableColumns.length > 0}
+          {#each workflows as workflow}
+            <tr
+              class="workflow-summary-configurable-row pinned"
+              on:click={() => goToWorkflow(workflow)}
+            >
+              {#if $supportsBulkActions}
+                <td>
+                  <Checkbox
+                    hoverable
+                    bind:group={selectedWorkflows}
+                    value={workflow}
+                    disabled={allSelected}
+                  />
+                </td>
+              {/if}
+              {#each pinnedColumns as column}
```

Comment:
If there are no pinned columns and the user drags/clicks a column to the top it will automatically be pinned. However, if there are no visible columns and a user adds a a column it will **not** be automatically pinned.

Just wanted to confirm this is expected behavior 📌